virttest/utils_split_daemons.py

```diff
@@ -0,0 +1,394 @@
+
+from virttest import libvirtd_decorator
+from virttest import remote
+from virttest import utils_misc
+from virttest.staging import service
+from virttest.utils_gdb import GDB
+
+
+class VirtDaemonCommon(object):
+
+    """
+    Common class to manage libvirt split daemon service on host or guest.
+    """
+    daemon_name = ''
+    virsh_cmd = "virsh -c qemu:///system list"
+
+    def __init__(self, daemon_name='', session=None):
+        """
+        Initialize an service object for virt daemons.
+
+        :param daemon_name: daemon name such as virtqemud, networkd,etc,.
+        :param session: An session to guest or remote host.
+        """
+        if daemon_name:
+            self.daemon_name = daemon_name
+
+        self.session = session
+        if self.session:
+            self.remote_runner = remote.RemoteRunner(session=self.session)
+            runner = self.remote_runner.run
+        else:
+            runner = process.run
+
+        if not self.daemon_name:
+            logging.warning("libvirt split daemon service is not available in host, "
+                            "utils_daemons module will not function normally")
+        self.virtdaemon = service.Factory.create_service(self.daemon_name, run=runner)
+
+    def _wait_for_start(self, timeout=60):
+        """
+        Wait n seconds for daemon to start. Default is 10 seconds.
+
+        :param timeout: time out for waiting start
+        """
+        def _check_start():
+            try:
+                if self.session:
+                    self.session.cmd(self.virsh_cmd, timeout=2)
+                else:
+                    process.run(self.virsh_cmd, timeout=2)
+                return True
+            except Exception:
+                return False
+        return utils_misc.wait_for(_check_start, timeout=timeout)
+
+    @libvirtd_decorator.libvirt_version_context_aware_libvirtd_split
+    def start(self, reset_failed=True):
+        if reset_failed:
+            self.virtdaemon.reset_failed()
+        if not self.virtdaemon.start():
+            return False
+        return self._wait_for_start()
+
+    @libvirtd_decorator.libvirt_version_context_aware_libvirtd_split
+    def stop(self):
+        return self.virtdaemon.stop()
+
+    @libvirtd_decorator.libvirt_version_context_aware_libvirtd_split
+    def restart(self, reset_failed=True):
+        if reset_failed:
+            self.virtdaemon.reset_failed()
+        if not self.virtdaemon.restart():
+            return False
+        return self._wait_for_start()
+
+    @libvirtd_decorator.libvirt_version_context_aware_libvirtd_split
+    def is_running(self):
+        return self.virtdaemon.status()
+
+
+class VirtQemud(VirtDaemonCommon):
+    """
+    Class for manage virtqemud daemon service on host or guest.
+    """
+    # Override daemon name in super class.
+    daemon_name = 'virtqemud'
+    # Override virsh command  in super class to verify daemon work.
+    virsh_cmd = "virsh -c qemu:///system list"
```

Comment:
As above

Reply:
see previous comment

Reply:
And also no inherit for parent __init__?

Reply:
remove this line since it is the same with parent class